core/include/kernel/interrupt.h

```diff
@@ -164,4 +197,148 @@ static inline struct itr_handler *itr_alloc_add(size_t it,
 				       0);
 }
 
+/*
+ * Interrupt controller chip API functions
+ */
+
+/*
+ * interrupt_call_handlers() - Call registered handlers for an interrupt
+ * @chip	Interrupt controller
+ * @itr_num	Interrupt number
+ *
+ * This function is called from an interrupt context by a primary interrupt
+ * handler. This function calls the handlers registered for that interrupt.
+ * If interrupt is not handled, it is masked.
+ */
+void interrupt_call_handlers(struct itr_chip *chip, size_t itr_num);
+
+/*
+ * interrupt_mask() - Mask an interrupt
+ * @chip	Interrupt controller
+ * @itr_num	Interrupt number
+ *
+ * This function may be called in interrupt context
+ */
+static inline void interrupt_mask(struct itr_chip *chip, size_t itr_num)
+{
+	chip->ops->mask(chip, itr_num);
+}
+
+/*
+ * interrupt_unmask() - Unmask an interrupt
+ * @chip	Interrupt controller
+ * @itr_num	Interrupt number
+ *
+ * This function may be called in interrupt context
+ */
+static inline void interrupt_unmask(struct itr_chip *chip, size_t itr_num)
+{
+	chip->ops->unmask(chip, itr_num);
+}
+
+/*
+ * interrupt_enable() - Enable an interrupt
+ * @chip	Interrupt controller
+ * @itr_num	Interrupt number
+ */
+static inline void interrupt_enable(struct itr_chip *chip, size_t itr_num)
+{
+	if (chip->ops->enable)
```

Comment:
Didn't we agree on making these mandatory?

Reply:
https://github.com/OP-TEE/optee_os/pull/5954#discussion_r1211795168:
> Why not make this function pointer mandatory? Same for the other ops below.

Right, I missed the last sentence, sorry. Ok to make also `.enable`/`.disable` mandatory.